code/game/alternate_appearance.dm

```diff
@@ -65,14 +65,15 @@ GLOBAL_LIST_EMPTY(active_alternate_appearances)
 	var/atom/target
 	var/image/image
 	var/add_ghost_version = FALSE
-	var/ghost_appearance
+	var/datum/atom_hud/alternate_appearance/basic/observers/ghost_appearance
 	uses_global_hud_category = FALSE
 
 /datum/atom_hud/alternate_appearance/basic/New(key, image/I, options = AA_TARGET_SEE_APPEARANCE)
 	..()
 	transfer_overlays = options & AA_MATCH_TARGET_OVERLAYS
 	image = I
 	target = I.loc
+	RegisterSignal(target, COMSIG_MOVABLE_Z_CHANGED, .proc/target_z_changed)
```

Comment:
[21:42:15] Runtime in stack_trace.dm, line 3: movable_ztransit overridden. Use override = TRUE to suppress this warning